components/app-core/backend/auth.go

```diff
@@ -257,12 +257,33 @@ func (p *portalProxy) logoutOfCNSI(c echo.Context) error {
 			"Need CNSI GUID passed as form param")
 	}
 
-	userID, err := p.GetSessionStringValue(c, "user_id")
+	userGUID, err := p.GetSessionStringValue(c, "user_id")
 	if err != nil {
-		return echo.NewHTTPError(http.StatusUnauthorized, "Could not find correct session value")
+		return fmt.Errorf("Could not find correct session value: %s", err)
 	}
 
-	p.deleteCNSIToken(cnsiGUID, userID)
+	cnsiRecord, err := p.GetCNSIRecord(cnsiGUID)
+	if err != nil {
+		return fmt.Errorf("Unable to load CNSI record: %s", err)
+	}
+
+	// If cnsi is cf AND cf is auto-register only clear the entry
+	if cnsiRecord.CNSIType == "cf" && p.GetConfig().AutoRegisterCFUrl == cnsiRecord.APIEndpoint.String() {
+		log.Info("Setting token record as disconnected")
```

Comment:
Can we change log.Info to log.Debug, otherwise we're going to see a lot of log junk.